From 99ea3798680459fe8f5259748f2b8f917f1c952e Mon Sep 17 00:00:00 2001
From: Alvaro Lopez Ortega <alvaro@gnu.org>
Date: Wed, 1 Jan 2014 17:18:01 +0100
Subject: [PATCH] Updates random number devices list.

The list of random number devices has been updated. Now /dev/hwrng is
tried first (hardware based generators interface) followed by
/dev/random and /dev/urandom.
---
 src/main.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/main.c b/src/main.c
index 19224ed..711ffe7 100644
--- a/src/main.c
+++ b/src/main.c
@@ -94,8 +94,9 @@ random_seed (void)
 	struct timeval tv;
 	unsigned int   seed;
 
-	if ((fd = open("/dev/urandom", O_RDONLY)) >= 0 ||
-	    (fd = open("/dev/random", O_RDONLY)) >= 0)
+	if ((fd = open("/dev/hwrng", O_RDONLY)) >= 0 ||
+	    (fd = open("/dev/random", O_RDONLY)) >= 0 ||
+	    (fd = open("/dev/urandom", O_RDONLY)) >= 0)
 	{
 		read (fd, &seed, sizeof(seed));
 		close (fd);
